pkg/controllers/disruption/consolidation.go

```diff
@@ -177,7 +182,39 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 
 	if allExistingAreSpot &&
 		results.NewNodeClaims[0].Requirements.Get(v1beta1.CapacityTypeLabelKey).Has(v1beta1.CapacityTypeSpot) {
-		if len(candidates) == 1 {
+		// Allow spot consolidation if SpotConsolidation featureflag is set to true by customers
+		if options.FromContext(ctx).FeatureGates.SpotConsolidation {
+			var allowConsolidation bool = true
+			// All possible replacements for the current candidate sorted by price
+			instanceTypesOrderedByPrice := results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions.OrderByPrice(results.NewNodeClaims[0].Requirements)
```

Comment:
Yes, NOT modifying the existing InstanceTypeOptions in the result is intentional. 

`instanceTypesOptionsFilteredByPrice` contains the instanceTypes that are lower priced than the current candidate. If we use this, we are bound to have a spot consolidation because these instanceTypes will not have the current candidate in the list.

Instead what we want to do is: Fetch all the possible InstanceTypes for a candidate from simulation, sort it by price and check if the current candidate is part of the first 15 instanceTypes window. If yes, do not consolidate. If no, consolidate by sending the filtered list of instanceTypes (`instanceTypesOptionsFilteredByPrice`) for launch because we want to replace the current candidate with the cheaper instanceType. I hope this explanation is clear.

Reply:
Ah yes, this makes sense. Thanks for the explanation. Is it worth adding more of this detail as comments in the code so it's easier for users who weren't part of the design talks to know a bit more of why these decisions were made?

Reply:
Added more details on the top of the function.